internal/sharedcomponent/sharedcomponent.go

```diff
@@ -27,19 +27,37 @@ func NewSharedComponents[K comparable, V component.Component]() *SharedComponent
 
 // GetOrAdd returns the already created instance if exists, otherwise creates a new instance
 // and adds it to the map of references.
-func (scs *SharedComponents[K, V]) GetOrAdd(key K, create func() (V, error)) (*SharedComponent[V], error) {
+func (scs *SharedComponents[K, V]) GetOrAdd(key K, create func() (V, error), telemetrySettings *component.TelemetrySettings) (*SharedComponent[V], error) {
 	if c, ok := scs.comps[key]; ok {
+		// If we haven't already seen this telemetry settings, this shared component represents
+		// another instance. Wrap ReportComponentStatus to report for all instances this shared
+		// component represents.
+		if _, ok := c.seenSettings[telemetrySettings]; !ok {
```

Comment:
I was being defensive about calling `GetOrAdd` multiple times. If this is something that we don't need to worry about it, I can remove the check (and `seenSettings`).

Reply:
In practice I don't think it's really intended that this function is called more than once per instantiation. That said, you're probably right to check it anyways because it is technically possible.